client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -69,10 +70,19 @@ export default React.createClass( {
 		analytics.pageView.record( '/checkout/domain-contact-information', 'Checkout > Domain Contact Information' );
 	},
 
-	validate( fieldNames, onComplete ) {
+	trimWhitespace( fieldValues, onComplete ) {
+		const trimmedFieldValues = Object.assign( {}, fieldValues );
+		this.fieldNames.forEach( ( fieldName ) => {
+			trimmedFieldValues[ fieldName ] = fieldValues[ fieldName ].trim();
```

Comment:
In case someone is wondering - if the field is empty, hidden, etc. it's still an empty string, so `fieldValues[ fieldName ]` should never be `undefined`.